clang/lib/AST/Expr.cpp

```diff
@@ -263,6 +263,14 @@ namespace {
   }
 }
 
+QualType Expr::getEnumCoercedType(const ASTContext &Ctx) const {
+  bool NotEnumType = dyn_cast<EnumType>(this->getType()) == nullptr;
+  if (NotEnumType)
```

Comment:
```suggestion
  if (!isa<EnumType>(this->getType()))
```

Though, would probably suggest just something like:
```
if (isa<EnumType>(getType())) return getType();

else if (const EnumConstantDecl *ECD = getEnumConstantDecl())
  return Ctx.getTypeDeclType...
return getType();

```

Reply:
done